pages/common/valgrind.md

```diff
@@ -0,0 +1,19 @@
+# valgrind
+
+> Wrapper for a set of expert tools for profiling, optimizing and debugging programs.
+
+- Test `program` with a specified Valgrind tool:
+
+`valgrind --tool={{tool_name}} {{program}}`
+
+- Use the (default) Memcheck tool to report all possible memory leaks of `program` in detail:
+
+`valgrind --leak-check=full --show-leak-kinds=all {{program}}`
```

Comment:
I don't think a general form is needed if we're providing two specific examples (cachegrind and massif). So my suggestion would be:

    valgrind {{program}}
    valgrind --leak-check=full --show-leak-kinds=all {{program}}
    valgrind --tool=cachegrind {{program}}
    valgrind --tool=massif --stacks=yes {{program}}